Formula/mpich.rb

```diff
@@ -54,8 +54,6 @@ def install
                           "--enable-shared",
                           "--enable-sharedlibs=gcc-osx",
                           "--with-pm=hydra",
-                          "CC=gcc-#{Formula["gcc"].any_installed_version.major}",
-                          "CXX=g++-#{Formula["gcc"].any_installed_version.major}",
                           "FC=gfortran-#{Formula["gcc"].any_installed_version.major}",
                           "F77=gfortran-#{Formula["gcc"].any_installed_version.major}",
```

Comment:
So this bakes `gfortran-XX` into the bottle when built. I don't know how the dep-tracking is done when the formula says `depends_on "gcc"`, but is there a situation where the user has `gcc@9` and the bottle is built with `gcc@10` and now the dependency is not correctly resolved? Perhaps homebrew-ites can clarify.

Reply:
I know that these lines aren't changed in this commit, but they were changed in the previous 3.4.1 update that introduced the `CC=...` lines

Reply:
Not sure if this addresses your concern, but on a machine with gcc and gcc@9 install, rebuilding the bottle will use gfortran-10. From the `gcc` requirement, I would assume that the bottle will not install directly if gcc@9 is present but gcc is not

Reply:
Yeah `depends_on "gcc"` always pulls in the _latest_ GCC.

If this is baked into the bottle and will break when `gcc` is updated to 11, then please add a test to check for this so it can be spotted.